exercise-solutions/tcp-server-exercises/src/bin/echo-log-scope.rs

```diff
@@ -0,0 +1,38 @@
+use std::{
+    io::{self, BufRead as _, BufReader, BufWriter, Write as _},
+    net::{TcpListener, TcpStream},
+    sync::{Mutex, MutexGuard},
+    thread,
+};
+
+fn handle_client(stream: TcpStream, log: &Mutex<Vec<usize>>) -> Result<(), io::Error> {
+    let reader = BufReader::new(&stream);
+    let mut writer = BufWriter::new(&stream);
+    for line in reader.lines() {
+        let line = line?;
+        {
+            let mut log: MutexGuard<'_, _> = log.lock().unwrap();
```

Comment:
The type annotation probably doesn't help here? rust-analyser will add it if you open the file, and by writing it in it looks like you have to write it in to make the code work, which isn't true.

Reply:
The idea was for it to be visible when reading code on GitHub, but I removed it.